src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1072,10 +1072,133 @@ public Document GetRevisionBefore(DocumentsOperationContext context, string id,
                 return result;
             }
         }
+        public async Task EnforceConfiguration()
+        {
+            var parameters = new Parameters
+            {
+                Before = DateTime.MinValue,
+                MinimalDate = DateTime.MinValue,
+                EtagBarrier = _documentsStorage.GenerateNextEtag(),
+                OnProgress = progress => { } 
+            };
+            parameters.LastScannedEtag = parameters.EtagBarrier;
+
+            var result = new OperationResult();
+            var ids = new List<LazyStringValue>();
+
+            var hasMore = true;
+            while (hasMore)
+            {
+                hasMore = false;
+                ids.Clear();
+
+                using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext ctx))
+                {
+                    using (ctx.OpenReadTransaction())
+                    {
+                        var revisions = new Table(RevisionsSchema, ctx.Transaction.InnerTransaction);
+                        foreach (var tvr in revisions.SeekBackwardFrom(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice],
+                            parameters.LastScannedEtag))
+                        {
+                            var state = ShouldProcessNextRevisionId(ctx, ref tvr.Reader, parameters, result, out var id);
+                            if (state == NextRevisionIdResult.Break)
+                                break;
+                            if (state == NextRevisionIdResult.Continue)
+                                continue;
+
+                            ids.Add(id);
+
+                            if (ids.Count > 1024 || ctx.AllocatedMemory > SizeLimit)
```

Comment:
Shouldn't we use 2 contexts like in RevertRevisions?

Reply:
Unlike the revert revisions we don't need to keep the entire revision, we need only the id.